zone/mob.cpp

```diff
@@ -8627,3 +8636,128 @@ void Mob::ClearDataBucketCache()
 		DataBucket::DeleteFromCache(id, t);
 	}
 }
+
+bool Mob::IsInGroupOrRaid(Mob* other, bool same_raid_group) {
+	if (!other || !IsOfClientBotMerc() || !other->IsOfClientBotMerc()) {
+		return false;
+	}
+
+	if (this == other) {
+		return true;
+	}
+
+	Raid* raid = IsBot() ? CastToBot()->GetStoredRaid() : (IsRaidGrouped() ? GetRaid() : nullptr);
+
+	if (raid) {
+		if (!other->IsRaidGrouped()) {
+			return false;
+		}
+
+		Raid* other_raid = other->IsBot() ? other->CastToBot()->GetStoredRaid() : other->GetRaid();
+
+		if (!other_raid) {
+			return false;
+		}
+
+		auto raid_group = raid->GetGroup(GetCleanName());
+		auto other_raid_group = other_raid->GetGroup(other->GetCleanName());
+
+		if (
+			raid_group == RAID_GROUPLESS || 
+			other_raid_group == RAID_GROUPLESS ||
+			(same_raid_group && raid_group != other_raid_group)
+		) {
+			return false;
+		}
+
+		return true;
+	}
+
+	Group* group = GetGroup();
+	Group* other_group = other->GetGroup();
+
+	return group && group == other_group;
+}
+
+bool Mob::DoLosChecks(Mob* other) {
+	if (!CheckLosFN(other) || !CheckWaterLoS(other)) {
+		if (CheckLosCheatExempt(other)) {
+			return true;
+		}
+		return false;
+	}
+
+	if (!CheckLosCheat(other)) {
+		return false;
+	}
+
+	return true;
+}
+
+bool Mob::CheckLosCheat(Mob* other) {
```

Comment:
Not really sure what all of this is trying to do differently than normal LosChecks, what is `CheckLosCheat` ?

Reply:
Door skipping by being able to send attacks, casts through doors. Some doors pass LoS checks but the client blocks it itself (I assume), so as a player it *can* be exploited but it's far easier with pets/bots. I have a separate PR that can address the client stuff if it's enabled once all this is done/in.